crates/driver/src/tests/setup/driver.rs

```diff
@@ -162,39 +149,60 @@ pub async fn setup(config: Config<'_>) -> Client {
         driver_addr,
         handle,
         match config.file {
-            ConfigFile::Create(_) => Some(config_file),
+            ConfigFile::Create { .. } => Some(config_file),
             ConfigFile::Load(_) => None,
         },
     )
 }
 
 /// Create the config file for the driver to use.
-async fn create_config_file(path: &ConfigPath, solvers: &[setup::Solver]) {
-    let configs = solvers
-        .iter()
-        .map(|solver| {
-            let setup::Solver {
-                config:
-                    setup::solver::Config {
-                        absolute_slippage,
-                        relative_slippage,
-                        address,
-                        name,
-                        ..
-                    },
-                addr,
-            } = solver;
-            #[rustfmt::skip]
+async fn create_config_file(
+    path: &ConfigPath,
+    solvers: &[setup::Solver],
+    contracts: &ContractsConfig,
+) {
+    #[rustfmt::skip]
+    let contracts_config = format!(
+        "[contracts]\n\
+        gp-v2-settlement = \"0x{}\"\n\
+        weth = \"0x{}\"",
+        hex::encode(contracts.gp_v2_settlement.unwrap().as_bytes()),
+        hex::encode(contracts.weth.unwrap().as_bytes()),
+    );
```

Comment:
```suggestion
    let contracts_config = format!(
        r#"
        [contracts]
        gp-v2-settlement = "0x{}"
        weth = "0x{}"
        "#,
        hex::encode(contracts.gp_v2_settlement.unwrap().as_bytes()),
        hex::encode(contracts.weth.unwrap().as_bytes()),
    );
```
This is a bit nicer. You can see the config file like you would when editing it yourself because there is no need for escaping. Also doesn't need rustfmt-skip.

Could create the full config this way too instead of joining the strings afterwards.

Reply:
Note that these aren't equivalent - the `r#` notation keeps leading white-space in the string (don't know if is an issue with TOML).

Reply:
That's true. I assumed TOML doesn't care about whitespace.

Reply:
> TOML doesn't care about whitespace.

It might be the case, just wanted to point it out. If it doesn't your suggestion definitely looks a lot nicer!

Reply:
https://github.com/cowprotocol/services/pull/1124/commits/7ebee4f0f543572dba7c785e43ef967a01bab393